app/views/unattended/provisioning_templates/snippet/ca_registration.erb

```diff
@@ -0,0 +1,39 @@
+<%#
+kind: snippet
+name: ca_registration
+model: ProvisioningTemplate
+snippet: true
+description: |
+  This template is used for updating Foreman's CA on hosts that are registered by Katello.
+  It replaces the CA used by subscription-manager and adds the CA to trusted anchors.
+-%>
+
+<% if plugin_present?('katello') -%>
+  # Define the path to the Katello server CA certificate
+  KATELLO_SERVER_CA_CERT=/etc/rhsm/ca/katello-server-ca.pem
+
+  # If katello ca cert file exists on host, update it and make sure it's in trust anchors
+  if [ -f "$KATELLO_SERVER_CA_CERT" ]; then
+    cat << EOF > "$KATELLO_SERVER_CA_CERT"
+<%= foreman_server_ca_cert %>
+EOF
+
+    if [ -f /etc/debian_version ]; then
+      CA_TRUST_ANCHORS=/usr/local/share/ca-certificates/
+    else
+      CA_TRUST_ANCHORS=/etc/pki/ca-trust/source/anchors
+    fi
+
+    # Add the Katello CA certificate to the system-wide CA certificate store
```

Comment:
Do we actually do this today? I don't see why we need to.

When we use custom certificates (like when Let's Encrypt is used as a CA) it should already be in the certificate store.

Perhaps this should also check if `$CA_TRUST_ANCHORS/katello-server-ca.pem` exists.

Reply:
yes, the katello-ca-consumer script does it for some reason today